Fix spelling

az0 opened the pull request:
Most of the changes are spelling fixes in code comments, and a few are corrections to grammar. Some changes are to documentation (e.g., Rmd) or to strings of log output (e.g., c_api.cc).

trivialfis reviewed (commented):
Thanks for the PR!  Could you please address some small comments?

az0 commented on doc/tutorials/rf.rst:
```diff
@@ -1,9 +1,9 @@
 #########################
-Random Forests in XGBoost
+Random Forests(TM) in XGBoost
```
Minitab owns a trademark for the term Random Forests ([source](https://en.wikipedia.org/wiki/Random_forest)). It's standard to write the trademark symbol once at the beginning of each document that mentions a trademarked term. When I read about [RST ](https://docutils.sourceforge.io/docs/ref/rst/directives.html), (TM) seems to convert to ™.

BTW, there's another XGBoost document that uses the ™ symbol with Random Forests, but it would be better to change it later so that it only uses the ™ symbol the first time.

trivialfis replied:
Thanks for the explanation.  Good to learn something new ;-)

az0 commented on doc/install.rst:
```diff
@@ -3,7 +3,7 @@ Installation Guide
 ##################
 
 XGBoost provides binary packages for some language bindings.  The binary packages support
-the GPU algorithm (``gpu_hist``) on machines with NVIDIA GPUs. Please note that **training
```
OK, I fixed this in a second commit, and I found two other cases like it. 

By the way, feel free to squash my two commits.

trivialfis commented:
Note to myself: make sure R documents are correctly generated.

trivialfis commented:
@az0 Thanks for the PR!  I fixed a small typo and updated the generated documents in R binding.

It would be great if in the future you can try to avoid using the master branch (or any other default branch name for that matter).  This will make it easier for others to push to your branch.  Just a minor suggestion.

az0 commented:
> @az0 Thanks for the PR! I fixed a small typo and updated the generated documents in R binding.

You are welcome. Thank you for quickly getting it merged.

> It would be great if in the future you can try to avoid using the master branch (or any other default branch name for that matter). This will make it easier for others to push to your branch. Just a minor suggestion.

Thank you for letting me know. I looked for instructions about this in the main README and the [XGBoost Community Guidelines](https://xgboost.readthedocs.io/en/latest/contrib/community.html), but I didn't see it. I'll keep it in mind for next time.

trivialfis commented:
@az0 

>  I looked for instructions about this in the main README and the XGBoost Community Guidelines

I think it's more a github workflow issue that applies to other projects too.  ;-)
